crates/partition-store/tests/snapshots_test/mod.rs

```diff
@@ -0,0 +1,75 @@
+use restate_partition_store::snapshots::{LocalPartitionSnapshot, PartitionSnapshotMetadata};
+use restate_partition_store::{PartitionStore, PartitionStoreManager};
+use restate_storage_api::fsm_table::{FsmTable, ReadOnlyFsmTable};
+use restate_storage_api::Transaction;
+use restate_types::config::WorkerOptions;
+use restate_types::identifiers::PartitionKey;
+use restate_types::live::Live;
+use restate_types::logs::Lsn;
+use std::ops::RangeInclusive;
+use tempfile::tempdir;
+use tracing::instrument;
+
+#[instrument(skip_all, level = "warn")]
+pub(crate) async fn run_tests(manager: PartitionStoreManager, mut partition_store: PartitionStore) {
+    insert_test_data(&mut partition_store).await;
+
+    let snapshots_dir = tempdir().unwrap();
+
+    let partition_id = partition_store.partition_id();
+    let path_buf = snapshots_dir.path().to_path_buf().join("sn1");
+
+    let snapshot = partition_store
+        .create_snapshot(path_buf)
+        .await
+        .unwrap();
+
+    let snapshot_meta = PartitionSnapshotMetadata {
+        cluster_name: "cluster_name".to_string(),
+        partition_id,
+        key_range: partition_store.partition_key_range().clone(),
+        min_applied_lsn: snapshot.min_applied_lsn,
+        files: snapshot.files.clone(),
+    };
+    let metadata_json = serde_json::to_string_pretty(&snapshot_meta).unwrap();
+    let db_comparator_name = snapshot.db_comparator_name.clone();
+
+    drop(partition_store);
+    drop(snapshot);
+    manager.drop_partition(partition_id).await;
```

Comment:
This is nasty but it allows me to demo that restoring basically works; I'll undo this in a later PR in this series.